### DISCUSSION TITLE
Current State node doesn't work as expected with home assistant.

### DISCUSSION THREAD

**Discussion by skurrasch** [zachowj/node-red-contrib-home-assistant-websocket#1720] [Q&A] [answered]
Hi,
I have a problem with a "Current state" node with home assistant.
When I inject the node with an "time stamp" node then everthing works as expected. The configured Entity ID is read and will be interpreted correctly.
If I use an "Event State" node then the ID of the event state node will be used for the if state in the "current state" node and not the configured ID.
Is this a bug or a misunderstanding myside?
Example:
`[{"id":"1926be33b2237dbe","type":"server-state-changed","z":"4af3e86983cd2abd","name":"Shelly BLU Button 1","server":"13dfd5af.c631ba","version":6,"outputs":1,"exposeAsEntityConfig":"","entities":{"entity":["event.shelly_blu_button1_c2d8_button_1"],"substring":[],"regex":[]},"outputInitially":false,"stateType":"str","ifState":"","ifStateType":"str","ifStateOperator":"is","outputOnlyOnStateChange":true,"for":"0","forType":"num","forUnits":"minutes","ignorePrevStateNull":false,"ignorePrevStateUnknown":false,"ignorePrevStateUnavailable":false,"ignoreCurrentStateUnknown":false,"ignoreCurrentStateUnavailable":false,"outputProperties":[{"property":"payload","propertyType":"msg","value":"","valueType":"eventData"},{"property":"topic","propertyType":"msg","value":"","valueType":"triggerId"}],"x":490,"y":280,"wires":[["d19038b3bc803119"]]},{"id":"d19038b3bc803119","type":"api-current-state","z":"4af3e86983cd2abd","name":"Value greater then","server":"13dfd5af.c631ba","version":3,"outputs":2,"halt_if":"20","halt_if_type":"num","halt_if_compare":"gt","entity_id":"sensor.shellyrolloterrasse_cover_0_power","state_type":"num","blockInputOverrides":false,"outputProperties":[{"property":"payload","propertyType":"msg","value":"","valueType":"entityState"},{"property":"data","propertyType":"msg","value":"","valueType":"entity"}],"for":"0","forType":"num","forUnits":"minutes","override_topic":false,"state_location":"payload","override_payload":"msg","entity_location":"data","override_data":"msg","x":790,"y":260,"wires":[["26f0f89d0b20c0e1"],["0e2f840ccd5a8f72"]]},{"id":"4dbffcab97f721fe","type":"inject","z":"4af3e86983cd2abd","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":520,"y":220,"wires":[["d19038b3bc803119"]]},{"id":"26f0f89d0b20c0e1","type":"debug","z":"4af3e86983cd2abd","name":"Greater","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":1020,"y":220,"wires":[]},{"id":"0e2f840ccd5a8f72","type":"debug","z":"4af3e86983cd2abd","name":"Smaller","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":1020,"y":300,"wires":[]},{"id":"13dfd5af.c631ba","type":"server","name":"Home Assistant","addon":true}]`

**Comment by romix92** [comment]
I have the same problem with nodes: device info -> current state (with if state).
State in current state is wrong.

**Comment by zachowj** [comment, marked as answer]
The current state nodes accept [input overrides](https://zachowj.github.io/node-red-contrib-home-assistant-websocket/node/current-state.html#inputs) via the message object. When `msg.payload.entityId` is set it overrides what is set in the local config of the node. You can either make sure the `payload` doesn't contain any properties that override the local config or check the box "[Block Input Overrides](https://zachowj.github.io/node-red-contrib-home-assistant-websocket/node/current-state.html#block-input-overrides)" in the current state node.

**Reply by skurrasch** [reply]
Great, that's it.
Thanks
